### ISSUE TITLE
Repeater items are not moved during the creation of a new resource

### ISSUE THREAD

**Issue by fenestron** [laravel/nova-issues#5837]
- Laravel Version: 10.19.0
- Nova Version: 4.27.12
- PHP Version: 8.2.9
- Database Driver & Version: PostgreSQL 15.4
- Operating System and Version: Ubuntu 22.04.3 LTS
- Browser type and version: Google Chrome 116.0.5845.96

### Description:

Repeater items are not moved during the creation of a new resource. However, they are moved during the update of the same resource.

[25.08.2023 14:22:51.webm](https://github.com/laravel/nova-issues/assets/9680621/79c4ad9b-3487-40e7-b44e-84c88d7ae749)

#### Model's fields:

```
public function fields(NovaRequest $request)
{
    return [
        ID::make()->sortable(),

        Repeater::make('Line Items', 'data')
            ->repeatables([
                LineItem::make(),
            ]),
    ];
}
```

#### LineItem.php
```
<?php

namespace App\Nova\Repeater;

use Laravel\Nova\Fields\Number;
use Laravel\Nova\Fields\Repeater\Repeatable;
use Laravel\Nova\Http\Requests\NovaRequest;

class LineItem extends Repeatable
{
    public function fields(NovaRequest $request)
    {
        return [
            Number::make('Quantity')->rules('required', 'numeric'),
        ];
    }
}
```

**Comment by crynobone** [comment]
Thank you for the report, we already noticed this and a fixed has been added pending next release.

**Comment by crynobone** [comment]
solved together with #5786
